### PR TITLE
Vertical tabs demo

### PR DESCRIPTION
**Pull request by Blackbaud-SteveBrush**

### CODE REVIEW

**Comment by Blackbaud-BobbyEarl** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-tabs?_sp.spa=pr-vertical-tabs-demo](https://host.nxt.blackbaud.com/skyux-tabs?_sp.spa=pr-vertical-tabs-demo)

**Comment by Blackbaud-SomaliaJamall** [comment]
@Blackbaud-AdamFunderburk The demo is in and merged with your code! Please review and edit in whatever ways need me, and let me know if the demo looks good.

File: src/app/docs/vertical-tabs/vertical-tabs-docs.component.html [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+  </sky-docs-demo>
+
+  <sky-docs-design-guidelines>
+    <sky-docs-demo-page-section heading="Usage">
+      <sky-docs-design-guideline heading="Use when" headingStyle="success">
+        <p>
+          Use vertical tabs when the number of tabs and/or the tab label text is expected to exceed the horizontal space
+          available.
+        </p>
+        <sky-docs-design-guideline-thumbnail
```
Comment (Blackbaud-ToddRoberts) [inline src/app/docs/vertical-tabs/vertical-tabs-docs.component.html:51]: this example looks like a split view to me

Comment (Blackbaud-AdamFunderburk) [reply src/app/docs/vertical-tabs/vertical-tabs-docs.component.html:51]: hmm, I can see that. Is it the header giving that impression? Should I try something less temporary and with less workflow implication?

Comment (Blackbaud-ToddRoberts) [reply src/app/docs/vertical-tabs/vertical-tabs-docs.component.html:51]: more the labels on the tabs. it reads like a list of items you are working through.

Comment (Blackbaud-AdamFunderburk) [reply src/app/docs/vertical-tabs/vertical-tabs-docs.component.html:51]: Do you think changing the label on the list itself can help? If we call these "Affiliates" or "Affiliated organizations", does the list appear more static and less like the kind of to-do list that would warrant a Split view?

**Comment by blackbaud-ado** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-tabs?_sp.spa=pr-vertical-tabs-demo](https://host.nxt.blackbaud.com/skyux-tabs?_sp.spa=pr-vertical-tabs-demo)

File: src/app/docs/vertical-tabs/vertical-tabs-docs.component.html [hunk 2]
```diff
@@ -0,0 +1,272 @@
+<sky-docs-demo-page moduleName="SkyTabsModule" moduleSourceCodePath="src/app/public/modules/vertical-tabset/"
+  pageTitle="Vertical tabs">
+  <sky-docs-demo-page-summary>
+    Vertical tabs divide and categorize page content when the number of categories is large.
+  </sky-docs-demo-page-summary>
+
+  <sky-docs-demo>
+    <sky-docs-demo-control-panel (selectionChange)="onDemoSelectionChangeTabs($event)">
+      <sky-docs-demo-control-panel-section>
+        <sky-docs-demo-control-panel-checkbox label="Include counts" propertyName="counts" [checked]="false">
```
Comment (blackbaud-johnly) [inline src/app/docs/vertical-tabs/vertical-tabs-docs.component.html:10]: The *Reset* option for both demos doesn't reset the selected state and collapsed/expanded state to its setting when the demo originally loaded. It only seems to reset whether to include the counts, but I'm pretty sure demos are supposed to reset everything back to the original state.

File: src/app/docs/vertical-tabs/vertical-tabs-docs.component.html [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+        </p>
+        <sky-docs-design-guideline-thumbnail
+          caption="Don't use vertical tabs when there is enough space for horizontal tabs." captionType="danger"
+          imageSource="~/assets/img/guidelines/vertical-tabs/vertical-tabs-usage-3.png">
+        </sky-docs-design-guideline-thumbnail>
+      </sky-docs-design-guideline>
+
+      <sky-docs-design-guideline>
+        <p>
+          Don’t combine vertical tabs and horizontal tabs. Change the workflow instead of nesting multiple sets of tabs.
```
Comment (blackbaud-johnly) [inline src/app/docs/vertical-tabs/vertical-tabs-docs.component.html:78]: Should we also have a "Don't use" guideline for not nesting vertical tabs within other vertical tabs? We have a guideline like that for horizontal tabs, and we hint at it here in the second sentence. It seems to me like we should have that as a distinct "Don't use" when item here.

Comment (Blackbaud-ToddRoberts) [reply src/app/docs/vertical-tabs/vertical-tabs-docs.component.html:78]: probably so, or maybe make the wording more generic to not next tabs in tabs (regardless of regular or vertical) and show both examples.

Comment (blackbaud-johnly) [reply src/app/docs/vertical-tabs/vertical-tabs-docs.component.html:78]: I'll try to go the more-generic route in my PR into this branch.

File: src/app/docs/vertical-tabs/vertical-tabs-docs.component.html (hunk 2 above)
Comment (blackbaud-johnly) [reply src/app/docs/vertical-tabs/vertical-tabs-docs.component.html:10]: @Blackbaud-SteveBrush and @Blackbaud-SomaliaJamall, do either of you remember what we changed in the other responsive demos for this behavior?

File: src/app/docs/vertical-tabs/vertical-tabs-docs.component.html (hunk 3 above)
Comment (blackbaud-johnly) [reply src/app/docs/vertical-tabs/vertical-tabs-docs.component.html:78]: I updated this in my branch where I'm adding descriptions for the dev properties and also implementing edits to the design guidelines. @Blackbaud-AdamFunderburk and @Blackbaud-ToddRoberts, if you guys can take a the PR for those changes at https://github.com/blackbaud/skyux-tabs/pull/83, I'd appreciate it. No rush though as I'm still trying to resolve questions about some dev properties.

File: src/app/docs/vertical-tabs/vertical-tabs-docs.component.html (hunk 2 above)
Comment (Blackbaud-SteveBrush) [reply src/app/docs/vertical-tabs/vertical-tabs-docs.component.html:10]: There’s a `(reset)` event that can be subscribed to, where the developer can then reset all the values needed. It’s not a fully automatic process.

File: src/app/docs/vertical-tabs/vertical-tabs-docs.component.html (hunk 1 above)
Comment (blackbaud-johnly) [reply src/app/docs/vertical-tabs/vertical-tabs-docs.component.html:50]: @Blackbaud-AdamFunderburk and @Blackbaud-ToddRoberts, can we resolve this conversation? Or are changes still pending?

Comment (Blackbaud-AdamFunderburk) [reply src/app/docs/vertical-tabs/vertical-tabs-docs.component.html:50]: I've updated the visual example vertical-tabs-usage-1

**Comment by blackbaud-ado** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-tabs?_sp.spa=pr-vertical-tabs-demo](https://host.nxt.blackbaud.com/skyux-tabs?_sp.spa=pr-vertical-tabs-demo)

**Comment by Blackbaud-SomaliaJamall** [comment]
I'm still working on the reset functionality, will have it done in a bit

**Comment by blackbaud-ado** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-tabs?_sp.spa=pr-vertical-tabs-demo](https://host.nxt.blackbaud.com/skyux-tabs?_sp.spa=pr-vertical-tabs-demo)

**Comment by blackbaud-ado** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-tabs?_sp.spa=pr-vertical-tabs-demo](https://host.nxt.blackbaud.com/skyux-tabs?_sp.spa=pr-vertical-tabs-demo)

File: src/app/docs/vertical-tabs/vertical-tabs-docs.component.html (hunk 2 above)
Comment (blackbaud-johnly) [reply src/app/docs/vertical-tabs/vertical-tabs-docs.component.html:10]: Looks like this is working as expected now. Thanks @Blackbaud-SomaliaJamall!

**Comment by blackbaud-ado** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-tabs?_sp.spa=pr-vertical-tabs-demo](https://host.nxt.blackbaud.com/skyux-tabs?_sp.spa=pr-vertical-tabs-demo)

**Pull request merged**